### PR TITLE
Vue envelope popover menu

### PR DESCRIPTION
**Pull request by ChristophWurst**
This basically re-implements https://github.com/nextcloud/mail/pull/973. cc @kevgathuku 

ToDo
- [x] Get structure right. I copied the Marionette one, which moves the menu into the details element, but that's incorrect according to the list standards at https://docs.nextcloud.com/server/14/developer_manual/design/list.html
- [x] ~~Switch to action component developed by @skjnldsv ~~
- [x] Extend folder service with a delete function
- [x] Remove envelope from folder list
- [x] Add remove transition

I would appreciate if @kevgathuku and/or @skjnldsv could help me with this. Somehow I can't get the structure right and it always looks awful.

### CODE REVIEW

**Comment by kevgathuku** [comment]
Hey @ChristophWurst do you mind rebasing/merging the latest version of the `refactor/vue` branch? 
Every time I switch to this branch I get the case insensitivity issues I talked about in https://github.com/nextcloud/mail/pull/1171

**Comment by ChristophWurst** [comment]
Done :)

**Comment by kevgathuku** [comment]
Thanks @ChristophWurst  👍

**Comment by kevgathuku** [comment]
Hey @ChristophWurst I'm trying to work on this and I'm facing an issue where I can't enable the mail app locally.

I'm following these steps:
1. Checkout latest nextcloud/server master branch commit 80f7add485
2. Clone the `mail` app in the apps folder master branch commit 4c2351d
3. Run `make dev-js` in the mail folder
4. Navigate to http://localhost:8000/index.php/settings/apps and enable the Mail app
5. Refresh the page, and the mail app is back on the bottom of the list, not enabled

I'm not sure if the issue is with the mail app or the server, so I'm not sure how to debug this issue. Any help would be greatly appreciated 😄

**Comment by ChristophWurst** [comment]
Anything in your nextcloud.log? The server disables apps if they throw exceptions.

**Comment by ChristophWurst** [comment]
No worries :) Thanks for helping with this :v:

**Comment by kevgathuku** [comment]
Hey @ChristophWurst another quick question :)
How do I load some emails to test with? 
The account create functionality is not there yet and I've been trying to get it to work but it will take some time.

**Comment by ChristophWurst** [comment]
> How do I load some emails to test with?
> The account create functionality is not there yet and I've been trying to get it to work but it will take some time.

Yeah that is :hankey: right now. You have to use master to set up the account(s) and then switch back to the vue branch.

**Comment by kevgathuku** [comment]
@ChristophWurst After looking into this some more I also can't find out what exactly is causing the menu not to be positioned correctly. 

I've added 1 commit with these changes: 
 - Fixing a minor structure issue i.e added the `app-content-list-item-menu` div after the `app-content-list-item-details`  div
 - Formatted the `Envelope.vue` file

The root of the issue seems to be the difference is the width of the `app-content-list-item-line-two` div in `master` and this branch but I can't trace where the difference is coming from 😞

**feature/vue-envelope-popover-menu**

screen shot 2018-10-08 at 13 23 17

**master**

screen shot 2018-10-08 at 13 24 10

**Comment by ChristophWurst** [comment]
^ @jancborchardt could you help us with this?

**Comment by ChristophWurst** [comment]
> What's left here? Only the mispositioned menu? :)

Yep!

**Comment by ChristophWurst** [comment]
Updated the branch to the latest changes on the Vue branch.

**Comment by skjnldsv** [comment]
Design fixed ;)

File: js/views/Home.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-	<div id="content" class="mail">
+	<div id="content" class="app-mail">
```
Comment (ChristophWurst) [inline js/views/Home.vue]: :heart:

**Comment by ChristophWurst** [comment]
Basic operations (toggle flagged, delete) work, including some collapse transitions. Edge cases like jumping to the next message when the currently active message is deleted does not work yet. This will befixed in a another PR.

**Pull request merged**